src/topp/CMakeLists.txt

```diff
@@ -41,8 +41,27 @@ cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
 # add all the tools
 include(executables.cmake)
 foreach(i ${TOPP_executables})
-  add_executable(${i} ${i}.cpp)
-  target_link_libraries(${i} OpenMS)
+  # support tools with subfolder or single cpp tools
+  if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${i})
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/${i})
+#[[
+# CMakeLists.txt in the subdirectory for NameOfTheTool cool look like this:
+cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
+project(NameOfTheTool)
+set(tool_sources
+  additional_file1.cpp
+  additional_file2.cpp
+  additional_header1.h
+  additional_header2.h
+)
+add_executable(${PROJECT_NAME} ${PROJECT_NAME}.cpp ${tool_sources}) # Add the main cpp file and additional files
+target_link_libraries(${PROJECT_NAME} OpenMS) # Link the executable with OpenMS
```

Comment:
You can add a if(NOT TARGET OpenMS) find_package(OpenMS) 

to allow building this tool separately from OpenMS

Reply:
ok do I understand this correctly that this - in principle, would allow us to build a topp tool with OpenMS as external dependency?

Reply:
quick questions for your opinion.
Would this also work, and would it be better? I see this quite often in other cmake projects
```
if(PROJECT_IS_TOP_LEVEL)
  find_package(OpenMS)
endif()
```